### ISSUE TITLE
Linear classification performance on CIFAR10/STL10 DataModules drops from Lightning 0.8.5/Bolts 0.1.0 to Lightning 0.9.0/Bolts 0.1.1

### ISSUE THREAD

**Issue by sachit-menon** [PyTorchLightning/pytorch-lightning-bolts#175]
## 🐛 Bug

Linear classification performance on CIFAR10/STL10 Bolts DataModules drops from Lightning 0.8.5/Bolts 0.1.0 to Lightning 0.9.0/Bolts 0.1.1. 

I know you do careful performance-matching testing for the different Lightning versions, so I thought I'd post this here too thinking it could be from something in the DataModules changing rather than Lightning. However, since switching Bolts versions gave errors without switching Lightning versions, I can't be certain about that. It could also be some change in defaults.

### To Reproduce

1. Train Moco on (unlabeled data from) STL10 DataModule with the Bolts implementation using resnet18 (I can provide a pretrained model of this part if it would be helpful)
2. Load trained model, freeze weights, set .eval()
3. Add a linear classifier on top of the penultimate layer representation, train with (labeled data from) STL10 DataModule with traditional self-supervised performance evaluation. (Train for 100 epochs with batch size of 256 and initial learning rate of 0.12; decay learning rate by a factor of 0.2 at epochs 60,80.)
4. Observe substantial drop in performance

On STL10, I am getting a performance drop of 16% top-1 validation accuracy (88->72), and a 0.5 increase in cross-entropy loss (0.3 to 0.8). On CIFAR10 I get a smaller, but consistent drop of 2% top-1 validation accuracy. Both of these are using the _exact same_ code, just with switching conda envs. (The reason for changing both the Lightning version and the Bolts version is that it seems required for compatibility between the two.)

#### Code sample
I can provide pretrained models on CIFAR10 and STL10 if that would be helpful, along with my full hyperparameter configs, but not the code at this time (sorry!). It's just a linear classifier though. 

### Expected behavior

Expected: same performance across versions.

### Environment

 - PyTorch Version (e.g., 1.0): 1.5.1 with the former/1.6.0 for the latter (compatibility)
 - OS (e.g., Linux): Linux
 - How you installed PyTorch (`conda`, `pip`, source): conda
 - Build command you used (if compiling from source):
 - Python version: 3.8.3
 - CUDA/cuDNN version:
 - GPU models and configuration: 2080Ti
 - Any other relevant information:

**Comment by Borda** [comment]
@annikabrundyn mind have a look 🐰

**Comment by ananyahjha93** [comment]
The reason there is a decline in performance of the linear classifier is that DataModules were initially mixing training and validation sets. Now we have made sure that we pass a fixed random seed to the random_split function in the datamodules.

**Comment by annikabrundyn** [comment]
@ananyahjha93 thanks for clearing this up - that makes a lot of sense. I'm going to close the issue then 👍
